metadata/src/main/java/com/redhat/lightblue/rest/metadata/AbstractMetadataResource.java

```diff
@@ -340,18 +288,54 @@ public String removeEntity(@Context SecurityContext sc, @PathParam(PARAM_ENTITY)
     public String clearDefaultVersion(@Context SecurityContext sc, @PathParam(PARAM_ENTITY) String entity) {
         Error.reset();
 
-        final Map<String, List<String>> mappedRoles = metadata.getMappedRoles();
+        try{
+            checkPermission(sc, MetadataRoles.UPDATE_DEFAULTVERSION);
+            return new SetDefaultVersionCommand(null, entity, null).execute();
+        }
+        catch(RoleException e){
+            throw new SecurityException("Unauthorized request for 'DELETE /?entity/default'.", e);
+        }
+    }
+
+    private void checkPermission(SecurityContext sc, MetadataRoles roleAllowed) throws RoleException{
+        final Map<MetadataRoles, List<String>> mappedRoles = metadata.getMappedRoles();
         if(mappedRoles == null || mappedRoles.size() == 0){
             // No authorization was configured
-            return new SetDefaultVersionCommand(null, entity, null).execute();
+            return;
         }
 
-        List<String> roles = mappedRoles.get(MetadataRoles.UPDATE_DEFAULTVERSION.toString());
-        for (String role : roles) {
+        List<String> roles = mappedRoles.get(roleAllowed);
+
+        if(roles.contains(MetadataConstants.ROLE_NOONE)){
+            throw new RoleException(roles);
+        }
+        else if(roles.contains(MetadataConstants.ROLE_ANYONE)){
+            return;
+        }
+
+        for(String role : roles){
             if (sc.isUserInRole(role)){
-                return new SetDefaultVersionCommand(null, entity, null).execute();
+                return;
             }
         }
-        throw new SecurityException("Unauthorized request for 'DELETE /?entity/default'. One of the following roles is needed: "+roles);
+
+        throw new RoleException(roles);
+    }
+
+    private static class RoleException extends Exception{
```

Comment:
The 3rd option was actually my initial implementation, but the problem there is that the calling method would not have access to the required roles for the SecurityException message (to maintain existing functionality). So instead I opted to have checkPermission simply throw the new RoleException. I am happy to make RoleException a proper public class, I didn't initially because of the specific use case.

I think ideally, the core framework would own the checkPermission logic (or some generic variation thereof) so that should we ever add a new "view" layer it would not need to re-implement this. As it is, the SecurityContext needs to be consulted and that seems to be a lightblue-rest element, so solving that seemed to be outside the scope of this issue. In this scenario, I think a RoleException probably would make more sense.